packages/eslint-plugin/src/rules/no-useless-constructor.ts

```diff
@@ -31,13 +31,15 @@ function checkAccessibility(node: TSESTree.MethodDefinition): boolean {
 }
 
 /**
- * Check if method is not unless due to typescript parameter properties
+ * Check if method is not unless due to typescript parameter properties and decorators
  */
 function checkParams(node: TSESTree.MethodDefinition): boolean {
   return (
     !node.value.params ||
     !node.value.params.some(
-      param => param.type === AST_NODE_TYPES.TSParameterProperty,
+      param =>
+        param.type === AST_NODE_TYPES.TSParameterProperty ||
+        param.decorators?.length,
```

Comment:
Aside: it might be the case that `param.decorators` would make sense as a check, as it might be that the parser doesn't create `param.decorators` if it would have length 0. But that's a nuance not yet strongly defined in AST parsing. #5020